Source/CombatExtended/CombatExtended/Jobs/WorkGiver_ModifyWeapon.cs

```diff
@@ -0,0 +1,261 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using RimWorld;
+using UnityEngine;
+using Verse;
+using Verse.AI;
+
+namespace CombatExtended
+{
+    public class WorkGiver_ModifyWeapon : WorkGiver_Scanner
```

Comment:
Please add a method like `// RimWorld.WorkGiver_DoBill private Job StartOrResumeBillJob(Pawn pawn, IBillGiver giver)` to send ticker job fail reasons like `JobFailReason.Is("UnderRequiredSkill".Translate(skillRequirement.minLevel), bill.Label);`  for lacking the right ingredients on the map.

Reply:
I've never done this before and i don't see any reason to do it now.